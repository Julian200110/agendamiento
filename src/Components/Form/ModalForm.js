import React, { useState } from 'react';
import { Row, Col, Button, Form } from 'react-bootstrap';
import '../../Stylesheets/Calendar/Calendar.css';
import Modal from 'react-bootstrap/Modal';
import PropTypes from 'prop-types';
import InputGroup from 'react-bootstrap/InputGroup';
import { UsePatients } from '../../Hooks/Patients/UsePatients.js';
import { useTranslation } from 'react-i18next';
import {
	handleCancelAppointment,
	handleConfirmAppointment,
	handleEditAppointment,
	handleCreateAppointment,
} from '../../Data/Appointment.js';
import { calculateAge } from '../../Helpers/DateHelper.js';
import PatientForm from './PatientForm';
import PersonalDataForm from './PersonalDataForm';
import ContactForm from './ContactForm';
import ReasonForm from './ReasonForm';
import DateTimePicker from './DateTimePicker';

function ModalForm({
	serviceData,
	modalShow,
	setModalShow,
	dueDate,
	setDueDate,
	selectedTime,
	setSelectedTime,
	selectedPatient,
	setSelectedPatient,
	setName,
	status,
	lastName,
	setLastName,
	documentType,
	setDocumentType,
	setDocument,
	document,
	dateOfBirth,
	setDateOfBirth,
	name,
	phoneNumber,
	setPhoneNumber,
	email,
	setEmail,
	reason,
	setReason,
	setStatus,
	idAppointment,
}) {
	const { patients } = UsePatients();
	const age = calculateAge(dateOfBirth, Date.now());
	const [isWhatsappChecked, setIsWhatsappChecked] = useState(true);
	const [isEmailChecked, setIsEmailChecked] = useState(true);
	const [isSmsChecked, setIsSmsChecked] = useState(true);
	const [t, i18n] = useTranslation('global');
	const documentTypeOptions = [
		'cc',
		'ce',
		'cd',
		'p',
		'sc',
		'pep',
		'rc',
		'ti',
		'cnv',
		'asi',
		'msi',
	];
	const handleSubmit = e => {
		e.preventDefault();
		handleCreateAppointment(setModalShow, selectedPatient);
	};
	return (
		<div>
			<Modal
				show={modalShow}
				onHide={() => setModalShow(false)}
				size='lg'
				aria-labelledby='contained-modal-title-vcenter'
				centered
				className='modal'
			>
				<Modal.Header closeButton>
					<Modal.Title id='contained-modal-title-vcenter'>
						{t('Codes.Request')}
					</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form onSubmit={handleSubmit}>
						<Row className='form-modal'>
							<Form.Group
								className='form-modal__group'
								controlId='formGrouService'
								change
							>
								<Form.Label className='form-modal__label'>
									{t('Codes.Service')}
								</Form.Label>
								<p className='form-modal__text'>
									{serviceData.service.code} - {serviceData.service.name} -{' '}
									{serviceData.selectedProfessional}
								</p>
							</Form.Group>
							<DateTimePicker
								dueDate={dueDate}
								setDueDate={setDueDate}
								selectedTime={selectedTime}
								setSelectedTime={setSelectedTime}
							/>
							<PatientForm
								selectedPatient={selectedPatient}
								setSelectedPatient={setSelectedPatient}
								patients={patients}
								setName={setName}
								setLastName={setLastName}
								setDocumentType={setDocumentType}
								setDocument={setDocument}
								status={status}
							/>
							{/* Nombre del paciente: se muestra solo si se selecciona 'Nuevo paciente' */}
							{selectedPatient === 'nuevoPaciente' && (
								<>
									<PersonalDataForm
										documentTypeOptions={documentTypeOptions}
										documentType={documentType}
										setDocumentType={setDocumentType}
										document={document}
										setDocument={setDocument}
										dateOfBirth={dateOfBirth}
										setDateOfBirth={setDateOfBirth}
										age={age}
										phoneNumber={phoneNumber}
										setPhoneNumber={setPhoneNumber}
										email={email}
										setEmail={setEmail}
										setName={setName}
										lastName={lastName}
										setLastName={setLastName}
										name={name}
									/>
									<ContactForm
										isWhatsappChecked={isWhatsappChecked}
										setIsWhatsappChecked={setIsWhatsappChecked}
										isEmailChecked={isEmailChecked}
										setIsEmailChecked={setIsEmailChecked}
										isSmsChecked={isSmsChecked}
										setIsSmsChecked={setIsSmsChecked}
									/>
								</>
							)}
							<ReasonForm
								status={status}
								reason={reason}
								setReason={setReason}
							/>{' '}
						</Row>
						<Row className='modal-footer'>
							<Button
								className='modal-footer__button modal-footer__button--close'
								onClick={() => {
									setModalShow(false);
									setStatus('new');
								}}
							>
								{t('Codes.buttons.Close')}
							</Button>
							{status === 'new' ? (
								<Button
									type='submit'
									className='modal-footer__button modal-footer__button--register'
								>
									{t('Codes.buttons.Register')}
								</Button>
							) : (
								<>
									<Button
										type='button'
										className='modal-footer__button modal-footer__button--cancel'
										onClick={() => {
											handleCancelAppointment(idAppointment, setModalShow);
										}}
									>
										{t('Codes.buttons.Cancel')}
									</Button>
									<Button
										type='button'
										className='modal-footer__button modal-footer__button--close'
										onClick={() => {
											handleConfirmAppointment(idAppointment, setModalShow);
										}}
									>
										{t('Codes.buttons.Confirm')}
									</Button>
									<Button
										type='button'
										className='modal-footer__button modal-footer__button--update'
										onClick={() => {
											handleEditAppointment(idAppointment, setModalShow);
										}}
									>
										{t('Codes.buttons.Update')}
									</Button>
								</>
							)}
						</Row>
					</Form>
				</Modal.Body>
			</Modal>
		</div>
	);
}
ModalForm.propTypes = {
	serviceData: PropTypes.any,
	modalShow: PropTypes.any,
	setModalShow: PropTypes.any,
	handleSubmit: PropTypes.any,
	dueDate: PropTypes.any,
	handleDateChangee: PropTypes.any,
	selectedTime: PropTypes.any,
	handleTimeChange: PropTypes.any,
	selectedPatient: PropTypes.any,
	handlePatientChange: PropTypes.any,
	patients: PropTypes.any,
	setName: PropTypes.any,
	status: PropTypes.any,
	lastName: PropTypes.any,
	setLastName: PropTypes.any,
	documentType: PropTypes.any,
	setDocumentType: PropTypes.any,
	setDocument: PropTypes.any,
	document: PropTypes.any,
	dateOfBirth: PropTypes.any,
	handleDateOfBirthChange: PropTypes.any,
	age: PropTypes.any,
	phoneNumber: PropTypes.any,
	setPhoneNumber: PropTypes.any,
	email: PropTypes.any,
	setEmail: PropTypes.any,
	reason: PropTypes.any,
	setReason: PropTypes.any,
	setStatus: PropTypes.any,
	handleCancelAppointment: PropTypes.any,
	handleConfirmAppointment: PropTypes.any,
	handleEditAppointment: PropTypes.any,
	name: PropTypes.any,
	setSelectedTime: PropTypes.any,
	setSelectedPatient: PropTypes.any,
	setDateOfBirth: PropTypes.any,
	setDueDate: PropTypes.any,
	idAppointment: PropTypes.any,
};

export default ModalForm;
