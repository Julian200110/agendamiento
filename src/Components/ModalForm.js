import React from 'react';
import { Row, Col, Button, Form } from 'react-bootstrap';
import '../Stylesheets/Calendar.css';
import Modal from 'react-bootstrap/Modal';
import PropTypes from 'prop-types';

import {
	handleCancelAppointment,
	handleConfirmAppointment,
	handleEditAppointment,
	handleCreateAppointment,
} from '../Data/Appointment';

function ModalForm({
	serviceData,
	modalShow,
	setModalShow,
	dueDate,
	setDueDate,
	selectedTime,
	setSelectedTime,
	selectedPatient,
	setSelectedPatient,
	patients,
	setName,
	status,
	lastName,
	setLastName,
	documentType,
	setDocumentType,
	setDocument,
	document,
	dateOfBirth,
	setDateOfBirth,
	name,
	age,
	phoneNumber,
	setPhoneNumber,
	email,
	setEmail,
	reason,
	setReason,
	setStatus,
	idAppointment,
}) {
	const handleSubmit = e => {
		e.preventDefault();
		handleCreateAppointment(setModalShow, selectedPatient);
	};
	return (
		<div>
			<Modal
				show={modalShow}
				onHide={() => setModalShow(false)}
				size='lg'
				aria-labelledby='contained-modal-title-vcenter'
				centered
				className='modal'
			>
				<Modal.Header closeButton>
					<Modal.Title id='contained-modal-title-vcenter'>
						Solicitud de cita
					</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form onSubmit={handleSubmit}>
						<Row className='form-modal'>
							<Form.Group
								className='form-modal__group'
								controlId='formGrouService'
								change
							>
								<Form.Label className='form-modal__label'>Servicio</Form.Label>
								<p className='form-modal__text'>
									{serviceData.service.code} - {serviceData.service.name} -{' '}
									{serviceData.selectedProfessional}
								</p>
							</Form.Group>
							<Form.Group
								className='form-modal__group'
								controlId='formGroupDate'
							>
								<Form.Label className='form-modal__label'>Fecha</Form.Label>
								<Form.Control
									type='date'
									className='form-modal__input'
									placeholder='Due date'
									defaultValue={dueDate}
									onChange={event => {
										setDueDate(event.target.value);
										event.preventDefault();
									}}
								/>
							</Form.Group>
							<Form.Group
								className='form-modal__group'
								controlId='formGroupTime'
							>
								<Form.Label className='form-modal__label'>Hora</Form.Label>
								<Form.Control
									type='time'
									className='form-modal__input'
									placeholder='Hora'
									value={selectedTime}
									onChange={e => {
										setSelectedTime(e.target.value);
									}}
								/>
							</Form.Group>
							<Form.Group
								className='form-modal__group'
								controlId='formGroupPatient'
							>
								<Form.Label className='form-modal__label'>Paciente</Form.Label>
								{status === 'exist' ? (
									<Form.Control
										type='text'
										value={selectedPatient}
										readOnly
										className='form-modal__input form-modal__input--read-only'
									/>
								) : (
									<Form.Select
										value={selectedPatient}
										className='form-modal__select'
										onChange={event => {
											setSelectedPatient(event.target.value); // Actualizar el estado cuando cambia la selección
											event.preventDefault();
										}}
									>
										<option value='blanco'> </option>
										{patients.map((opcion, index) => (
											<option key={opcion.id} value={opcion}>
												{opcion.nameFirst} {opcion.nameSecond}{' '}
												{opcion.surnameFirst} {opcion.surnameSecond}
											</option>
										))}
										<option value='nuevoPaciente'>Nuevo paciente</option>
									</Form.Select>
								)}
							</Form.Group>
							{/* Nombre del paciente: se muestra solo si se selecciona 'Nuevo paciente' */}
							{selectedPatient === 'nuevoPaciente' && (
								<>
									<Form.Group
										className='form-modal__group'
										controlId='formGroupPatientName'
									>
										<Form.Label className='form-modal__label'>
											Nombre del paciente
										</Form.Label>
										<Form.Control
											className='form-modal__input'
											placeholder='Nombre(s)'
											value={name}
											onChange={e => setName(e.target.value)}
										/>
									</Form.Group>
									<Form.Group
										className='form-modal__group'
										controlId='formGroupPatientLastName'
									>
										<Form.Label className='form-modal__label'>
											Apellido del paciente
										</Form.Label>
										<Form.Control
											className='form-modal__input'
											placeholder='Apellido(s)'
											value={lastName}
											onChange={e => setLastName(e.target.value)}
										/>
									</Form.Group>
									<Row className='form-modal__row'>
										<Form.Group as={Col} controlId='formGroupTypeDocument'>
											<Form.Label className='form-modal__label'>
												Tipo de Documento
											</Form.Label>
											<Form.Select
												value={documentType}
												className='form-modal__select'
												onChange={e => setDocumentType(e.target.value)}
											>
												<option value='blanco'> </option>
												<option value='blanco'> </option>
												<option value='cc'>Cédula de ciudadanía</option>
												<option value='ce'>Cédula de extranjería</option>
												<option value='cd'>Carnet diplomático</option>
												<option value='p'>Pasaporte</option>
												<option value='s'>Salvoconucto</option>
												<option value='pep'>
													Permiso Especial de Permanencia
												</option>
												<option value='rc'>Registro civil</option>
												<option value='ti'>Tarjeta de identidad</option>
												<option value='cnv'>Certificiado de nacido vivo</option>
												<option value='ai'>Adulto sin identificar</option>
												<option value='mi'>Menor sin identificar</option>
											</Form.Select>
										</Form.Group>

										<Form.Group as={Col} controlId='formGroupDocument'>
											<Form.Label className='form-modal__label'>
												Documento de Identificación
											</Form.Label>
											<Form.Control
												className='form-modal__input'
												placeholder='Documento'
												value={document}
												onChange={e => setDocument(e.target.value)}
											/>
										</Form.Group>
									</Row>
									<Row className='form-modal__row'>
										<Form.Group as={Col} controlId='formGroupDateOfBirth'>
											<Form.Label className='form-modal__label'>
												Fecha de nacimiento
											</Form.Label>
											<Form.Control
												type='date'
												className='form-modal__input'
												value={dateOfBirth}
												onChange={event => {
													setDateOfBirth(event.target.value);
												}}
											/>
										</Form.Group>

										<Form.Group as={Col} controlId='formGroupAge'>
											<Form.Label className='form-modal__label'>
												Edad
											</Form.Label>
											<Form.Control
												type='text'
												value={age + ' años'}
												readOnly
												className='form-modal__input form-modal__input--read-only'
											/>
										</Form.Group>
									</Row>
									<Row className='form-modal__row'>
										<Form.Group as={Col} controlId='formGroupPhoneNumber'>
											<Form.Label className='form-modal__label'>
												Teléfono Celular
											</Form.Label>
											<Form.Control
												className='form-modal__input'
												placeholder='Teléfono celular'
												value={phoneNumber}
												onChange={e => setPhoneNumber(e.target.value)}
											/>
										</Form.Group>
										<Form.Group as={Col} controlId='formGroupEmail'>
											<Form.Label className='form-modal__label'>
												Email
											</Form.Label>
											<Form.Control
												type='email'
												className='form-modal__input'
												placeholder='Email'
												value={email}
												onChange={e => setEmail(e.target.value)}
											/>
										</Form.Group>
									</Row>
								</>
							)}
							<Form.Group
								className='form-modal__group'
								controlId='formGroupReason'
							>
								<Form.Label className='form-modal__label'>
									Motivo de consulta
								</Form.Label>
								{status === 'exist' ? (
									<Form.Control
										as='textarea'
										value={reason}
										readOnly
										className='form-modal__input form-modal__input--read-only'
									/>
								) : (
									<Form.Control
										as='textarea'
										className='form-modal__textarea'
										placeholder='Motivo'
										value={reason}
										onChange={e => setReason(e.target.value)}
									/>
								)}
							</Form.Group>
						</Row>
						<Row className='modal-footer'>
							<Button
								className='modal-footer__button modal-footer__button--close'
								onClick={() => {
									setModalShow(false);
									setStatus('new');
								}}
							>
								Cerrar
							</Button>
							{status === 'new' ? (
								<Button
									type='submit'
									className='modal-footer__button modal-footer__button--register'
								>
									Registrar
								</Button>
							) : (
								<>
									<Button
										type='button'
										className='modal-footer__button modal-footer__button--cancel'
										onClick={() => {
											handleCancelAppointment(idAppointment, setModalShow);
										}}
									>
										Cancelar
									</Button>
									<Button
										type='button'
										className='modal-footer__button modal-footer__button--close'
										onClick={() => {
											handleConfirmAppointment(idAppointment, setModalShow);
										}}
									>
										Confirmar
									</Button>
									<Button
										type='button'
										className='modal-footer__button modal-footer__button--update'
										onClick={() => {
											handleEditAppointment(idAppointment, setModalShow);
										}}
									>
										Actualizar
									</Button>
								</>
							)}
						</Row>
					</Form>
				</Modal.Body>
			</Modal>
		</div>
	);
}
ModalForm.propTypes = {
	serviceData: PropTypes.any,
	modalShow: PropTypes.any,
	setModalShow: PropTypes.any,
	handleSubmit: PropTypes.any,
	dueDate: PropTypes.any,
	handleDateChangee: PropTypes.any,
	selectedTime: PropTypes.any,
	handleTimeChange: PropTypes.any,
	selectedPatient: PropTypes.any,
	handlePatientChange: PropTypes.any,
	patients: PropTypes.any,
	setName: PropTypes.any,
	status: PropTypes.any,
	lastName: PropTypes.any,
	setLastName: PropTypes.any,
	documentType: PropTypes.any,
	setDocumentType: PropTypes.any,
	setDocument: PropTypes.any,
	document: PropTypes.any,
	dateOfBirth: PropTypes.any,
	handleDateOfBirthChange: PropTypes.any,
	age: PropTypes.any,
	phoneNumber: PropTypes.any,
	setPhoneNumber: PropTypes.any,
	email: PropTypes.any,
	setEmail: PropTypes.any,
	reason: PropTypes.any,
	setReason: PropTypes.any,
	setStatus: PropTypes.any,
	handleCancelAppointment: PropTypes.any,
	handleConfirmAppointment: PropTypes.any,
	handleEditAppointment: PropTypes.any,
	name: PropTypes.any,
	setSelectedTime: PropTypes.any,
	setSelectedPatient: PropTypes.any,
	setDateOfBirth: PropTypes.any,
	setDueDate: PropTypes.any,
	idAppointment: PropTypes.any,
};

export default ModalForm;
