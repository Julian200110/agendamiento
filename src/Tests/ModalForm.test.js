/* eslint-disable no-undef */
import React from 'react'; // Importa React
import { render, fireEvent, screen } from '@testing-library/react';
import ModalForm from '../Components/Form/ModalForm';
import '@testing-library/jest-dom';

test('changes due date and selected time', () => {
	const setSelectedTime = jest.fn();
	const setDueDate = jest.fn();
	const setModalShow = jest.fn();
	const setStatus = jest.fn();
	const setSelectedPatient = jest.fn();
	const serviceData = {
		service: {
			id: '25',
			code: '101',
			name: 'General adultos',
			modes: 'Extramural - Domiciliario',
			duration: 15,
			color: '#FF4445',
		},
		selectedProfessional: 'Juan',
	};
	const component = render(
		<ModalForm
			serviceData={serviceData}
			dueDate={'2023-10-10'}
			patients={[
				{
					nameFirst: 'Jhon',
					nameSecond: 'Doe',
					surnameFirst: 'Joe',
					surnameSecond: 'Done',
					id: 1,
				},
			]}
			status={'new'}
			lastName={'Jhon'}
			documentType={'cc'}
			document={1111111}
			dateOfBirth={'2001-10-10'}
			name={'Jhon'}
			phoneNumber={'3104444444'}
			email={'jhondoe@gmail.com'}
			reason={'Fever'}
			idAppointment={1}
			setSelectedTime={setSelectedTime}
			setDueDate={setDueDate}
			setModalShow={setModalShow}
			modalShow={true}
			setStatus={setStatus}
			setSelectedPatient={setSelectedPatient}
		/>,
	);
	expect(component).toMatchSnapshot();
	const newDueDate = '2023-11-01';
	const newTime = '10:00';

	// Change the due date
	fireEvent.change(screen.getByLabelText('Fecha'), {
		target: { value: newDueDate },
	});

	// Change the selected time
	fireEvent.change(screen.getByLabelText('Hora'), {
		target: { value: newTime },
	});

	expect(screen.getByText('Solicitud de cita')).toBeInTheDocument();
	fireEvent.click(screen.getByText('Cerrar'));
});
