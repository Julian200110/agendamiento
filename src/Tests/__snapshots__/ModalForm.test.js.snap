// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes due date and selected time 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div>
      <div />
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-labelledby="contained-modal-title-vcenter"
      aria-modal="true"
      class="fade modal modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
              id="contained-modal-title-vcenter"
            >
              Solicitud de cita
            </div>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body"
          >
            <form
              class=""
            >
              <div
                class="form-modal row"
              >
                <div
                  class="form-modal__group"
                >
                  <label
                    class="form-modal__label form-label"
                    for="formGrouService"
                  >
                    Servicio
                  </label>
                  <p
                    class="form-modal__text"
                  >
                    101
                     - 
                    General adultos
                     -
                     
                    Juan
                  </p>
                </div>
                <div
                  class="form-modal__group"
                >
                  <label
                    class="form-modal__label form-label"
                    for="formGroupDate"
                  >
                    Fecha
                  </label>
                  <input
                    class="form-modal__input form-control"
                    id="formGroupDate"
                    placeholder="Due date"
                    type="date"
                    value="2023-10-10"
                  />
                </div>
                <div
                  class="form-modal__group"
                >
                  <label
                    class="form-modal__label form-label"
                    for="formGroupTime"
                  >
                    Hora
                  </label>
                  <input
                    class="form-modal__input form-control"
                    id="formGroupTime"
                    placeholder="Hora"
                    type="time"
                    value=""
                  />
                </div>
                <div
                  class="form-modal__group"
                >
                  <label
                    class="form-modal__label form-label"
                    for="formGroupPatient"
                  >
                    Paciente
                  </label>
                  <select
                    class="form-modal__select form-select"
                    id="formGroupPatient"
                  >
                    <option
                      value="blanco"
                    >
                       
                    </option>
                    <option
                      value="1"
                    >
                      Jhon
                       
                      Doe
                       
                      Joe
                       
                      Done
                    </option>
                    <option
                      value="nuevoPaciente"
                    >
                      Nuevo paciente
                    </option>
                  </select>
                </div>
                <div
                  class="form-modal__group"
                >
                  <label
                    class="form-modal__label form-label"
                    for="formGroupReason"
                  >
                    Motivo de consulta
                  </label>
                  <textarea
                    class="form-modal__textarea form-control"
                    id="formGroupReason"
                    placeholder="Motivo"
                  >
                    Fever
                  </textarea>
                </div>
              </div>
              <div
                class="modal-footer row"
              >
                <button
                  class="modal-footer__button modal-footer__button--close btn btn-primary"
                  type="button"
                >
                  Cerrar
                </button>
                <button
                  class="modal-footer__button modal-footer__button--register btn btn-primary"
                  type="submit"
                >
                  Registrar
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
